### PR TITLE
feat(compounds): Add tests for buttons and icons

### PR DESCRIPTION
**Pull request by umahaea**
Introduce compounds into ux-test-platform, so created necessary place holders.
Add tests for buttons and icons (both desktop and mobile)
Add some common methods for file operations

@StommePoes @JasonLantz @mhomolak @wyseguyonline @udhadpa for your review.

### CODE REVIEW

File: src/main/shell_scripts/components.sh [hunk 1]
```diff
@@ -16,6 +16,27 @@ cp -R ~/build/Pearson-Higher-Ed/ux-test-platform/elements/dist/fonts ~/build/Pea
...
+git clone https://github.com/Pearson-Higher-Ed/compounds.git
+cd compounds
+git checkout $1
+npm install &>/dev/null
+npm run copy-utils
+npm run build &>/dev/null
+cp ~/build/Pearson-Higher-Ed/ux-test-platform/compounds/build/dist.compounds.js ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/compounds/jsfiles/
+cp -R ~/build/Pearson-Higher-Ed/ux-test-platform/compounds/node_modules/pearson-elements/dist/fonts ~/build/Pearson-Higher-Ed/ux-test-platform/
+cp -R ~/build/Pearson-Higher-Ed/ux-test-platform/compounds/node_modules/pearson-elements/dist/icons ~/build/Pearson-Higher-Ed/ux-test-platform/
+#cp ~/build/Pearson-Higher-Ed/ux-test-platform/compounds/node_modules/pearson-elements/dist/css/elements.css ~/build/Pearson-Higher-Ed/ux-test-platform/src/main/java/compounds/css/
```
Comment (umahaea) [inline src/main/shell_scripts/components.sh]: I will uncomment this line, as soon as we release new minor version of elements. Until then, manually installing elements here. The below lines will be gone after elements release.

File: src/main/java/compounds/fixtures/buttons.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+</head>
+<body>
+
+<body>
+<h2>Compounds - Buttons</h2>
+<div>
+    <div id="button-target">
+    </div>
+</div>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/react/15.4.2/react.js"></script>
```
Comment (StommePoes) [inline src/main/java/compounds/fixtures/buttons.html]: Question to everyone, do these tests need to pull in React/DOM like this or do they get them by running Compounds themselves anyway?

File: src/test/java/compoundsTests/ButtonsTest.java [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+        if (!isHeight) {
+            log.info("height for button type " + btnType + " is not as per the spec, actual: " + height);
+        }
+        commonUtils.writeInitialConfig(tempJSFilePath, buttonsJSFilePath);
+        Assert.assertTrue(isBackgroundColor && isFontSize && isLineHeight && isHeight);
+    }
+
+    @Test(testName = "Verify click operation on the Button Test", groups = "desktop-regression")
+    private void clickEventButtonTest() throws InterruptedException {
+        if (browser.equals("safari")) {
```
Comment (StommePoes) [inline src/test/java/compoundsTests/ButtonsTest.java]: is there something other than an alert that can be used to test button clickiness?

**Comment by StommePoes** [comment]
Other than my comments, looks great.

File: src/test/java/compoundsTests/ButtonsTest.java (hunk 3 above)
Comment (umahaea) [inline src/test/java/compoundsTests/ButtonsTest.java]: there can be many, I chose "alert".. as it is more interactive :)

File: src/main/java/compounds/fixtures/buttons.html (hunk 2 above)
Comment (JasonLantz) [inline src/main/java/compounds/fixtures/buttons.html]: that's how they pull them in currently... I'm looking at building vendor libs for the next version...

**Comment by umahaea** [comment]
Ok, got enough approvals. merging it.

**Pull request merged**